### PR TITLE
Feature cov287

### PR DESCRIPTION
**Pull request by Roleen20**
Refactor of sms service
Fixing logic of the wallet created via mobile session

### CODE REVIEW

File: V1/Controllers/WalletController.cs [hunk 1]
```diff
@@ -27,9 +27,10 @@ public WalletController(IWalletService walletService)
         /// <param name="walletParameters"></param>
         /// <returns></returns>
         [HttpPost]
-        public async Task<IActionResult> CreateWallet(CreateWalletRequest walletParameters)
+        [Route("{sessionId}")]
```
Comment (deltaDirk) [inline V1/Controllers/WalletController.cs:30]: This should not be a route, it should just be a query parameter

File: V1/Controllers/WalletController.cs [hunk 2]
```diff
@@ -27,9 +27,10 @@ public WalletController(IWalletService walletService)
...
+        public async Task<IActionResult> CreateWallet([FromBody] CreateWalletRequest walletParameters, string sessionId = null)
```
Comment (deltaDirk) [inline V1/Controllers/WalletController.cs:31]: [FromQuery] sessionId

Comment (deltaDirk) [reply V1/Controllers/WalletController.cs:31]: Also doesn't have to be defaulted to null, strings are automatically null if nothing is present.

File: V1/Interfaces/Services/ISmsService.cs [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 using System.Threading.Tasks;
-using CoviIDApiCore.V1.Constants;
 using CoviIDApiCore.V1.DTOs.SMS;
 
 namespace CoviIDApiCore.V1.Interfaces.Services
 {
     public interface ISmsService
     {
-        Task<SmsResponse> SendMessage(string mobileNumber, DefinitionConstants.SmsType smsType, string organisation = null, DateTime expireAt = default, Guid sessionId = default);
+        Task<SmsResponse> SendMessage(string mobileNumber);
```
Comment (deltaDirk) [inline V1/Interfaces/Services/ISmsService.cs:9]: This name is ambiguous. What message does it send?

File: V1/Interfaces/Services/IWalletService.cs [hunk 4]
```diff
@@ -2,13 +2,14 @@
...
 using CoviIDApiCore.Models.Database;
+using System;
 
 namespace CoviIDApiCore.V1.Interfaces.Services
 {
     public interface IWalletService
     {
         Task<WalletStatusResponse> GetWalletStatus(string walletId, string key);
-        Task<TokenResponse> CreateWalletAndOtp(CreateWalletRequest walletRequest);
+        Task<TokenResponse> CreateWalletAndOtp(CreateWalletRequest walletRequest, string sessionId = null);
```
Comment (deltaDirk) [inline V1/Interfaces/Services/IWalletService.cs:12]: Same comment as on the controller

File: V1/Services/SmsService.cs [hunk 5]
```diff
@@ -62,10 +43,19 @@ public async Task<SmsResponse> SendMessage(string mobileNumber, SmsType smsType,
                 ValidityPeriod = validityPeriod
             };
         }
+        public async Task SendWelcomeSms(string mobileNumber, string organisationName, DateTime expireAt, Guid sessionId)
+        {
+            var url = _configuration.GetValue<string>("WebsiteDomian");
+            url = $"{url}/create=wallet/details?sessionId={sessionId}";
```
Comment (deltaDirk) [inline V1/Services/SmsService.cs:49]: Save this partial route under the UrlConstants file please

File: V1/Controllers/WalletController.cs (hunk 2 above)
Comment (Roleen20) [reply V1/Controllers/WalletController.cs:31]: It's there for readability purposes

**Pull request merged**